plugins/InspectionGadgets/testsrc/com/siyeh/ig/fixes/controlflow/ExpressionMayBeFactorizedFixTest.java

```diff
@@ -174,6 +171,32 @@ public void testDoNotCleanupWithXOrOperator() {
                                "}\n");
   }
 
+  public void testDoNotCleanupWithFinalLazyActiveExpression() {
+    assertQuickfixNotAvailable(InspectionGadgetsBundle.message("if.may.be.factorized.quickfix"),
+                               "class X {\n" +
+                               "  boolean test() {\n" +
+                               "    boolean duplicateExpression = true;\n" +
+                               "    boolean thenExpression = true;\n" +
+                               "    boolean updatedExpression = true;\n" +
+                               "    boolean newValue = false;\n" +
+                               "    boolean evaluation = (duplicateExpression || thenExpression) &&/**/ (duplicateExpression || updatedExpression = newValue);\n" +
```

Comment:
Here also red code in the test data. Intentional?

Reply:
It wasn't intentional. I fixed it.